app/src/marketplace/containers/ProjectEditing/DataUnionSecrets/SecretList.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { FunctionComponent } from 'react'
+import styled from 'styled-components'
+import { DataUnionSecret } from '$app/src/marketplace/types/project-types'
+import { SecretItem } from './SecretItem'
+
+const Container = styled.div`
+    display: grid;
+    grid-gap: 12px;
+    margin: 32px 0;
+`
+
+type Props = {
+    secrets: Array<DataUnionSecret>,
+    onEdit: (secret: DataUnionSecret) => void,
+    onDelete: (secret: DataUnionSecret) => void,
+}
+
+export const SecretList: FunctionComponent<Props> = ({ secrets, onEdit, onDelete }: Props) => {
+    if (secrets == null || secrets.length === 0) {
```

Comment:
Yeah, maybe the null check could go but that `length === 0` will definitely happen and we don't want to render empty grid for nothing :)

Reply:
I'd consider skipping the whole component outside. I'm thinking:

```tsx
interface Props {
    secrets: { 0: DataUnionSecret } & DataUnionSecret[],
    // ...
}

function SecretList({ secrets }) {
    // Ts ensures at least 1 element. Otherwise redness!
}

// Outside

<>
    {secrets?.length > 0 && <SecretList secrets={secrets} />}
</>
```

Just an idea for a component that's easier to grasp at a glance.